packages/test/test-end-to-end-tests/src/test/nodeDidNotRunGc.spec.ts

```diff
@@ -0,0 +1,277 @@
+		[],
+		[],
+		[],
+		createDataStoreRuntime(),
+	);
+	const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+		runtime.IFluidHandleContext.resolveHandle(request);
+	const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+		[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		[dataStoreFactory2.type, Promise.resolve(dataStoreFactory2)],
+	]);
+
+	const createContainer = async (
+		disableSummary: boolean = true,
+		logger?: ITelemetryLogger,
+	): Promise<IContainer> => {
+		let summaryConfigOverrides: ISummaryConfiguration;
+		if (disableSummary) {
+			summaryConfigOverrides = { state: "disabled" };
+		} else {
+			const IdleDetectionTime = 100;
+			summaryConfigOverrides = {
+				...DefaultSummaryConfiguration,
+				...{
+					maxIdleTime: IdleDetectionTime,
+					maxTime: IdleDetectionTime * 12,
+					initialSummarizerDelayMs: 0,
+				},
+			};
+		}
+		const runtimeOptions: IContainerRuntimeOptions = {
+			summaryOptions: {
+				summaryConfigOverrides,
+			},
+		};
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+			runtimeOptions,
+		);
+		return provider.createContainer(runtimeFactory, { logger });
+	};
+
+	async function waitForSummaryOp(container: IContainer) {
+		await new Promise<void>((resolve) => {
+			container.deltaManager.on("op", (op: ISequencedDocumentMessage) => {
+				if (op.type === MessageType.Summarize) {
+					resolve();
+				}
+			});
+		});
+	}
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+	});
+
+	itExpects(
+		"summarizeOnDemand should fail with NodeDidNotRunGC error",
+		[
+			{
+				eventName: "fluid:telemetry:SummarizerNode:NodeDidNotRunGC",
+				clientType: "noninteractive/summarizer",
+			},
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel",
+				clientType: "noninteractive/summarizer",
+			},
+		],
+		async () => {
+			const mainContainer = await createContainer();
+			const mainDataStore = await requestFluidObject<TestDataObject1>(
+				mainContainer,
+				"default",
+			);
+			mainDataStore._root.set("mode", "write");
+			await waitForContainerConnection(mainContainer);
+
+			const { summarizer } = await createSummarizerFromFactory(
+				provider,
+				mainContainer,
+				dataStoreFactory1,
+				undefined,
+				undefined,
+				registryStoreEntries,
+			);
+
+			const result = summarizer.summarizeOnDemand({ reason: "test" });
+			const submitResult = await result.summarySubmitted;
+			assert(!submitResult.success, "The summary should have failed");
+			assert.strictEqual(
+				submitResult.stage,
+				"generate", // This would be "upload" without this change.
+				"Summary should have failed after generate",
+			);
+			assert.strictEqual(
+				submitResult.error.message,
+				"NodeDidNotRunGC",
+				"Summary failed with unexpected error",
+			);
+		},
```

Comment:
Would it be interesting to also reload the container and assert that there's an unreferenced node?

Reply:
@agarwal-navin

Reply:
This node created during summarize will not be part of the GC data and won't change the node count because of [this code](https://github.com/microsoft/FluidFramework/blob/fef3d20033faad2a596a59dd27bef303ec108d4d/packages/runtime/container-runtime/src/dataStores.ts#L714).
If the handle is stored in a DDS and then removed, then it will end up unreferenced which is not happening in this test.

Reply:
Oh right, we're not using bindToContext